### ISSUE TITLE
TLS termination option when using AWS ELB?

### ISSUE THREAD

**Issue by StaymanHou** [deis/workflow#267]
Hi. I tried to setup TLS termination when using AWS ELB together with AWS Certificate Manager. I changed the ELB port 443 listener, assigned the ssl certificate managed with AWS Certificate Manager. After a few minutes, the cert start working correctly. However, after a few hours/days, the ELB listeners setting got reset. And it start to deliver the self-signed cert generated by deis workflow.

Wonder if there's an option to disable ELB auto reset behaviour and tell deis workflow that we are using ELB as a TLS termination proxy?

**Comment by StaymanHou** [comment]
ping https://github.com/kubernetes/kubernetes/issues/24978

**Comment by Joshua-Anderson** [comment]
I'm super sorry nobody got back to you about this! I suspect this has to be supported first at the kuberenetes level, because it's the one that creates the elb. The router chart could probably support an option to automatically enable/disable ssl elb termination once kubernetes supports it though. I'm not experienced with the router however, so I'll let someone else more experienced with the router follow up.

**Comment by StaymanHou** [comment]
@Joshua-Anderson . Thank you very much for the update!

**Comment by bacongobbler** [comment]
Hi @StaymanHou, I'm terribly sorry nobody got to this ticket. This is most definitely the case that Kubernetes is overriding the TLS certificate, as the kubelets manage the load balancer and constantly call reconvergence on the load balancers' config.

In more recent Kubernetes releases, you are able to attach certificates to the ELB through the `service.beta.kubernetes.io/aws-load-balancer-ssl-certificate`.

I'm going to close this issue as this is an upstream issue and there is nothing actionable we can do here, and it looks like this was fixed up in Kubernetes v1.3. Thanks!

**Comment by panchr** [comment]
Are there any guides on how to set up TLS termination with ELB? Using `kubectl --namespace=deis edit service deis-route`, I've added the following to the annotations for the router:

```
    service.beta.kubernetes.io/aws-load-balancer-connection-idle-timeout: "1200"
    service.beta.kubernetes.io/aws-load-balancer-ssl-cert: {ARN of ACM certificate}
    service.beta.kubernetes.io/aws-load-balancer-ssl-ports: https
```

When I go to `https://app.example.com`, I get a 400 error: "The plain HTTP request was sent to HTTPS port". Then, adding this annotation: `service.beta.kubernetes.io/aws-load-balancer-backend-protocol: https`, I get a 404 error (presumably with the app not found).

Is there a list of annotations I need to add to enable TLS termination on ELB? I want platform-level TLS, such that all apps (and the Deis management endpoint, `deis.example.com`) to support HTTPS. There is a lot of fragmented documentation about this, but nothing definitive that I've found.
